src/dotnet/CommandFactory/CommandResolution/LocalToolsCommandResolver.cs

```diff
@@ -50,6 +50,33 @@ public CommandSpec Resolve(CommandResolverArguments arguments)
             return resolveResult;
         }
 
+        public CommandSpec Resolve(CommandResolverArguments arguments)
+        {
+            if (arguments == null || string.IsNullOrWhiteSpace(arguments.CommandName))
+            {
+                return null;
+            }
+
+            if (!arguments.CommandName.StartsWith(LeadingDotnetPrefix, StringComparison.OrdinalIgnoreCase))
+            {
+                return null;
+            }
+
+            var resolveResultWithoutLeadingDotnet = GetPackageCommandSpecUsingMuxer(arguments,
```

Comment:
Could this be written as:

```c#
...
// Try resolving without prefix first
var result = GetPackageCommandSpecUsingMuxer(arguments, new ToolCommandName(arguments.CommandName.Substring(LeadingDotnetPrefix.Length)));
if (result != null)
{
    return result;
}

// Fallback to resolving with the prefix
return GetPackageCommandSpecUsingMuxer(arguments, new ToolCommandName(arguments.CommandName));
```

To me, it appears that `resolveResultWithoutLeadingDotnet` is being returned if not null, regardless of `resolveResultWithLeadingDotnet` being null or not.  Therefore, what I suggest seems to be equivalent and only resolves once if we can resolve without the prefix rather than always twice.

Reply:
used your version